### PR TITLE
feat(metrics): Add scopes to Glean access token check event

### PR DESCRIPTION
**Pull request by vbudhram**
## Because

- We want to see the scopes in our Glean events for access token checked

## This pull request

- Adds the scopes as a sorted, comma separated string to those events

## Issue that this pull request solves

Closes: https://mozilla-hub.atlassian.net/browse/FXA-10484

## Checklist

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Other information (Optional)

@chenba You are probably the best to review this. I wasn't sure if this was the correct way to add an extra value to Glean events. I didn't want to piggy back off the `reason` value since it didn't seem to make sense in this case.

### CODE REVIEW

File: packages/fxa-shared/metrics/glean/fxa-backend-metrics.yaml [hunk 1]
```diff
@@ -414,6 +414,10 @@ access_token:
     expires: never
     data_sensitivity:
       - interaction
+    extra_keys:
+      scopes:
+        description: The scopes of the access token
+        type: string
```
Comment (vbudhram) [inline packages/fxa-shared/metrics/glean/fxa-backend-metrics.yaml:420]: Does Glean support Array types? Would that allow for easier sorting and graphs?

Comment (chenba) [reply packages/fxa-shared/metrics/glean/fxa-backend-metrics.yaml:420]: Unfortunately, the String List type is not support for the  `extra keys`.  😞

**Comment by chenba** [comment]
@bbangert @vbudhram what do you think of filtering out the common scopes (e.g. 'profile')?

**Review by chenba — APPROVED** [review]
Thanks, @vbudhram.  LGTM but let's see what Ben think of filtering out the common  scopes.  I think the intention is to use the scopes to understand what product/client is making the request; the common scopes don't really help with that.

**Review by bbangert — APPROVED** [review]
LGTM!

**Comment by bbangert** [comment]
The common scopes don't help, but I don't think they'll hinder either. However they could result in an increase in cardinality, so filtering out the profile/identity/openid basic ones should be done.

**Comment by bbangert** [comment]
I don't think filtering out the common scopes will matter here, mainly since I believe we'll be doing more specific queries in Looker/etc to quantify the other resource requests. The list of FxA handled scopes can vary greatly over time as well, so keeping those up to date and mapped to 'external' resource servers or not will probably be more time intensive than just adding a Looker view to see access token issuance for all tokens with scope including 'relay', etc.

**Comment by vbudhram** [comment]
> Do we have a white-list of all valid scope strings we verify this input on? e.g. this scope set data isn't completely unfiltered client-sent data?

Looks like we do store a list in the [database](https://github.com/mozilla/fxa/blob/b41e7756e8ce1d2d2726ee365422a46d4bc87cbf/packages/fxa-auth-server/lib/oauth/grant.js#L92) and request checks that the scope is valid, so in theory should only see valid scopes in Looker.

**Pull request merged**